fury/ui.py

```diff
@@ -1973,11 +2014,21 @@ def format_text(self):
     def update(self):
         """ Updates the slider. """
         # Compute the ratio determined by the position of the slider disk.
-        length = float(self.right_x_position - self.left_x_position)
-        if length != self.track.width:
-            raise Exception("Disk position outside the slider line")
-        disk_position_x = self.handle.center[0]
-        self._ratio = (disk_position_x - self.left_x_position) / length
+        disk_position_x = None
+        disk_position_y = None
+
+        if self.axis == 0:
+            length = float(self.right_x_position - self.left_x_position)
+            if length != self.track.width:
+                raise Exception("Disk position outside the slider line")
```

Comment:
`Exception` is too generic, you should pick another one

Reply:
You mean the Type of Exception right?
for e.g:
`raise HandlePositionBeyondSlider("Exception Text")`

Reply:
yes, exactly. Not a custom one but just a standard one like `ValueError`